cibuildwheel/windows.py

```diff
@@ -189,10 +187,28 @@ def setup_python(
     # Install pip
 
     requires_reinstall = not (installation_path / "Scripts" / "pip.exe").exists()
+
     if requires_reinstall:
         # maybe pip isn't installed at all. ensurepip resolves that.
         call(["python", "-m", "ensurepip"], env=env, cwd=CIBW_INSTALL_PATH)
 
+    # pip bundled with python 3.10 for arm64 lacks windows launcher and requires an extra pip force-reinstall to get pip executable
+    if python_configuration.arch == "ARM64" and Version(get_pip_version(env)) < Version("21.3"):
+        call(
+            [
+                "python",
+                "-m",
+                "pip",
+                "install",
+                "--force-reinstall",
+                "--upgrade",
+                "pip",
+                *dependency_constraint_flags,
+            ],
+            env=env,
+            cwd=CIBW_INSTALL_PATH,
+        )
+
```

Comment:
This might be overcomplicated. If the pip executable is missing, we should already detect that. Could you try looking for whatever you're missing, like-

```
if python_configuration.arch == "ARM64" and not (installation_path / "Scripts" / "piparm64.exe").exists():
    requires_reinstall = True
```

or just  this would be okay for now I think-

```
if python_configuration.arch == "ARM64":
    requires_reinstall = True
```

probably better to move this up to after the requires_reinstall set line too.

Reply:
That's exactly what I tried at first but unfortunately didn't work.

The issue is that pip bundled with python 3.10 builds executables such as pip.exe for x64 platform. The first re-install of pip updates pip module but builds pip.exe using the old pip which still generates x64 executable. But the second re-install uses updated pip and correctly builds pip.exe for the target.

Reply:
ah, hmm, okay. I think I understand. What is the `Version(get_pip_version(env)) < Version("21.3")` about? And are you assuming that requires_reinstall is always true on arm64?

Reply:
I'd expect this is a temporary problem and should be fixed once a newer pip is bundled? That's a guess, though, and it would be nice to verify this really was fixed in 21.3 and is not just a problem with nuget.

Reply:
Yes, the fix is available from pip release 21.3 so we don't have to do the extra install.

Reply:
Got it. I've suggested adding this explanation as a comment, which will help let us know when it can be removed.